crates/rpc/src/handlers/block.rs

```diff
@@ -0,0 +1,48 @@
+use alloy_primitives::B256;
+use ream_consensus::deneb::beacon_block::BeaconBlock;
+use ream_storage::{db::ReamDB, tables::Table};
+use serde::{Deserialize, Serialize};
+use tree_hash::TreeHash;
+use warp::{
+    http::status::StatusCode,
+    reject::Rejection,
+    reply::{Reply, with_status},
+};
+
+use super::BeaconResponse;
+use crate::types::{errors::ApiError, id::ID};
+
+#[derive(Serialize, Deserialize)]
+struct BlockRootResponse {
+    pub root: B256,
+}
+
+/// Called by `/blocks/<block_id>/root` to get the Tree hash of the Block.
+pub async fn get_block_root(block_id: ID, db: ReamDB) -> Result<impl Reply, Rejection> {
+    let block = get_block_from_id(block_id, &db).await?;
+    Ok(with_status(
+        BeaconResponse::json(BlockRootResponse {
+            root: block.tree_hash_root(),
+        }),
+        StatusCode::OK,
+    ))
+}
+
+/// Get the Block from provided Block ID
+pub async fn get_block_from_id(block_id: ID, db: &ReamDB) -> Result<BeaconBlock, ApiError> {
+    let block_id = match block_id {
+        ID::Slot(slot) => db.slot_index_provider().get(slot),
+        ID::Root(root) => Ok(Some(root)),
+    }
+    .map_err(|_| ApiError::InternalError)?
+    .ok_or(ApiError::NotFound(format!(
+        "Failed to find `block_root` from {block_id:?}"
+    )))?;
+
+    db.beacon_block_provider()
+        .get(block_id)
+        .map_err(|_| ApiError::InternalError)?
+        .ok_or(ApiError::NotFound(format!(
+            "Failed to find `beacon_state` from {block_id:?}"
+        )))
+}
```

Comment:
we don't need to fetch the block from the database, because the indexes give it to us no? So fetching the block and recomputing the tree_hash_root is a little redundent in this example